cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -61,6 +61,7 @@ func (dp *dotnetProject) Restore(
 ) *async.TaskWithProgress[*ServiceRestoreResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServiceRestoreResult, ServiceProgress]) {
+			task.SetProgress(NewServiceProgress("Restoring .NET dependencies"))
```

Comment:
Yes, this is to support a consistent inner loop experience and consistency with other languages/frameworks.  Once it gets to build/package it won't need to restore since it previously completed this step.